packages/mds-webpack-config/index.ts

```diff
@@ -118,16 +118,22 @@ const MergeConfigurations = (config: CustomConfiguration, name: string, path?: s
   )
 }
 
-const ConfigMethods = (name: string, path?: string) => ({
-  UsingDefaultConfig: () => MergeConfigurations({}, name, path),
-  UsingCustomConfig: (config: CustomConfiguration) => MergeConfigurations(config, name, path)
-})
+class WebpackConfigurationBuilder {
+  private name: string
+
+  constructor(private path: string, name?: string) {
+    this.name = name || parse(path).name
+  }
+
+  public UsingDefaultConfig() {
```

Comment:
What's the distinction between a default config and a custom config?

Reply:
A custom config allows overriding the standard webpack config with additional settings, plugins, etc. that are specifically required by the bundle being generated.